internal/source/prometheus/config.go

```diff
@@ -0,0 +1,43 @@
+package prometheus
+
+import (
+	promApi "github.com/prometheus/client_golang/api/prometheus/v1"
+	"gopkg.in/yaml.v3"
+
+	"github.com/kubeshop/botkube/pkg/api/source"
+	"github.com/kubeshop/botkube/pkg/ptr"
+)
+
+// Config prometheus configuration
+type Config struct {
+	URL             string               `yaml:"url"`
+	AlertStates     []promApi.AlertState `yaml:"alertStates"`
+	IgnoreOldAlerts *bool                `yaml:"ignoreOldAlerts"`
+}
+
+func MergeConfigs(configs []*source.Config) (Config, error) {
+	out := Config{
+		URL:             "http://localhost:9090",
+		AlertStates:     []promApi.AlertState{promApi.AlertStateFiring, promApi.AlertStatePending, promApi.AlertStateInactive},
+		IgnoreOldAlerts: ptr.Bool(true),
+	}
+	for _, rawCfg := range configs {
+		var cfg Config
+		err := yaml.Unmarshal(rawCfg.RawYAML, &cfg)
+		if err != nil {
+			return Config{}, err
+		}
+
+		if cfg.URL != "" {
+			out.URL = cfg.URL
+		}
+		if len(cfg.AlertStates) > 0 {
+			out.AlertStates = cfg.AlertStates
```

Comment:
WDYT about appending them instead of resetting 🤔 ? 
you can have a `map[string]struct{}` to make sure that you have only unique `AlertStates`.

In general, I'm fine with both approaches, but I'm curious why you picked overriding instead of merging 🙂

Reply:
Because, think about we have defaults "pending", "firing" in our values yaml. Then customer wants to see only "inactive". In this case, if I append instead of reset, then we will be showing alerts that have state that customer does not want.